src/main.cpp

```diff
@@ -2883,6 +2886,13 @@ bool ConnectBlock(const CBlock& block, CValidationState& state, CBlockIndex* pin
 
     SaplingMerkleTree sapling_tree;
     assert(view.GetSaplingAnchorAt(view.GetBestAnchor(SAPLING), sapling_tree));
+    // We don't perform the same check that Sprout does against an old tree root
+    // here because Sapling does not have interstitial tree states.
```

Comment:
The reason we're not doing this check has nothing to do with interstitial tree states. In fact, I'm not sure why we're not doing the check for Sapling except that it's just a performance thing. I would even advocate just removing the check for Sprout, since I think it was just intended as a way to find out if we screwed up our changes to the `CCoins` part of the code.